src/model/Publication.ts

```diff
@@ -229,7 +229,7 @@ export default class Publication extends R2Publication {
         // maybe in the future this can be done in the background?
         const result = await fetch(href);
         const length = (await result.blob()).size;
-        link.contentLength = length;
+        (link as Link).contentLength = length;
```

Comment:
Removing the `Link` type on L46 is what made it necessary to type cast this if I'm not wrong

Reply:
because this didn't actually work, not sure why, but the `contentLength` and `contentWeight` were simply always undefined